src/kernels/MIOpenInterpolate.cpp

```diff
@@ -0,0 +1,982 @@
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+#ifndef MIOPEN_DONT_USE_HIP_RUNTIME_HEADERS
+#include <hip/hip_fp16.h>
+#include <hip/hip_runtime.h>
+#endif
+
+#include "float_types.h"
+#include "tensor_view.hpp"
+
+__device__ inline FLOAT_ACCUM compute_linear_scale_factor(FLOAT_ACCUM scale_factor,
+                                                          int64_t input_size,
+                                                          int64_t output_size,
+                                                          bool align_corners)
+{
+    if(align_corners)
+    {
+        if(input_size == 1)
+        {
+            return static_cast<FLOAT_ACCUM>(output_size);
+        }
+        return static_cast<FLOAT_ACCUM>(output_size - 1) / (input_size - 1);
+    }
+    else if(scale_factor == 0)
+    {
+        return static_cast<FLOAT_ACCUM>(output_size) / input_size;
+    }
+    else
+    {
+        return static_cast<FLOAT_ACCUM>(scale_factor);
+    }
+}
+
+__device__ inline FLOAT_ACCUM
+get_src_index(int64_t dest_index, FLOAT_ACCUM scale_factor, bool align_corners)
+{
+    if(align_corners)
+    {
+        return dest_index / scale_factor;
+    }
+    else
+    {
+        // Follow Opencv resize logic.
+        return (dest_index + 0.5f) / scale_factor - 0.5f;
+    }
+}
+
+__device__ inline int64_t
+linear_back_index(int64_t src, FLOAT_ACCUM scale_factor, bool align_corners)
+{
+    return static_cast<int64_t>(ceil(get_src_index(src, 1.f / scale_factor, align_corners)));
+}
+
+__device__ inline void compute_linear_back_index_from_to(int64_t src,
+                                                         int64_t input_size,
+                                                         int64_t output_size,
+                                                         FLOAT_ACCUM scale_factor,
+                                                         bool align_corners,
+                                                         int64_t* from,
+                                                         int64_t* to)
+{
+    if(src - 1 < 1)
+    {
+        *from = 0;
+    }
+    else
+    {
+        *from = linear_back_index(src - 1, scale_factor, align_corners);
+    }
+    if(src + 1 > input_size)
+    {
+        *to = output_size;
+    }
+    else
+    {
+        *to = min(output_size, linear_back_index(src + 1, scale_factor, align_corners));
+    }
+}
+
+__device__ inline void compute_source_index_and_lambda(int64_t h,
+                                                       FLOAT_ACCUM scale_factor,
+                                                       int64_t Hin,
+                                                       int64_t Hout,
```

Comment:
Do you need `Hout` in this function?

Reply:
I kept it as in modnn, not sure other Ops using it or another reason

Reply:
Actually I checked this function in modnn and it does not have Hout, thats why I found this 😂😂

Reply:
![image](https://github.com/user-attachments/assets/de4c27c8-9a1c-42a3-81f5-f8929e3881d7)
hmmm

Reply:
Hmm so strange. Will look at it again tomorrow. Get some sleep buddy 😘

Reply:
Can't sleep, miss you so much babe

Reply:
Oh they removed it in a commit in May
![Uploading IMG_7685.png…]()

Reply:
> Can't sleep, miss you so much babe

😵‍💫😵‍💫

Reply:
fixed it